File: app/src/main/java/in/kashewdevelopers/airdistance/SplashScreen.java
package in.kashewdevelopers.airdistance;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import in.kashewdevelopers.airdistance.databinding.ActivitySplashScreenBinding;

public class SplashScreen extends AppCompatActivity {

    ActivitySplashScreenBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySplashScreenBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.markerPath.setVisibility(View.INVISIBLE);
        binding.appName.setVisibility(View.INVISIBLE);

        Animation animation = AnimationUtils.loadAnimation(this, R.anim.animate_marker);
        binding.marker.startAnimation(animation);

        animation = AnimationUtils.loadAnimation(this, R.anim.animate_path);
        binding.markerPath.startAnimation(animation);

        animation = AnimationUtils.loadAnimation(this, R.anim.animate_path);
        binding.appName.startAnimation(animation);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(SplashScreen.this, MapsActivity.class));
            }
        }, 2000);
    }

}

File: app/src/main/java/in/kashewdevelopers/airdistance/SplashScreen.kt
package `in`.kashewdevelopers.airdistance

import `in`.kashewdevelopers.airdistance.databinding.ActivitySplashScreenBinding
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.View
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity

class SplashScreen : AppCompatActivity() {

    private lateinit var binding: ActivitySplashScreenBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashScreenBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.markerPath.visibility = View.INVISIBLE
        binding.appName.visibility = View.INVISIBLE

        var animation = AnimationUtils.loadAnimation(this, R.anim.animate_marker)
        binding.marker.startAnimation(animation)

        animation = AnimationUtils.loadAnimation(this, R.anim.animate_path)
        binding.markerPath.startAnimation(animation)

        animation = AnimationUtils.loadAnimation(this, R.anim.animate_path)
        binding.appName.startAnimation(animation)

        Handler(Looper.getMainLooper())
                .postDelayed({ startActivity(Intent(this@SplashScreen, AirDistance::class.java)) }, 2000)
    }

}
